lib/containers/runtime.pm

```diff
@@ -0,0 +1,96 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test Docker’s btrfs storage driver features for image and container management
+# Maintainer: qac team <qa-c@suse.de>
+
+package containers::runtime;
+use Mojo::Base -base;
+use testapi;
+use Test::Assert 'assert_equals';
+
+has runtime => undef;
+
+sub _rt_assert_script {
+    my ($self, $cmd, @args) = @_;
+    assert_script_run($self->runtime . " " . $cmd, @args);
+}
+
+sub _rt_script_run {
+    my ($self, $cmd, @args) = @_;
+    return script_run($self->runtime . " " . $cmd, @args);
+}
+
+sub _rt_script_output {
+    my ($self, $cmd, @args) = @_;
+    return script_output($self->runtime . " " . $cmd, @args);
+}
+
+sub build {
+    my ($self, $dockerfile_path, $container_tag) = @_;
+    die 'wrong number of arguments' if @_ < 3;
+    assert_script_run "cd $dockerfile_path";
+    $self->_rt_assert_script("build -t $container_tag .");
+    record_info "$container_tag created";
+}
+
+sub up {
+    my ($self, $tag, %args) = @_;
+    die 'tag is required' unless $tag;
+    my $mode   = $args{daemon} ? '-d'    : '-it';
+    my $remote = $args{cmd}    ? 'sh -c' : '';
+    my $ret    = $self->_rt_script_run(sprintf qq(run --rm %s %s %s '%s'), $mode, $tag, $remote, $args{cmd});
+    record_info "Remote run on $tag", $args{cmd};
+    return $ret;
+}
+
+sub pull {
+    my ($self, $tag) = @_;
+    my $ret = $self->_rt_script_run("pull $tag");
+    return $ret;
+}
+
+sub retrieve_images {
```

Comment:
I would have called it `list_images` or `get_images`, but I'm also fine with this name.

Reply:
FYI whenever it is possibly i try to avoid `get_` prefix(or `get_`). About the `list_images` no problem from my side. But i will leave it for now if it is not a big deal and i will have it in mind in the next PR